### ISSUE TITLE
BulkAction Authorization return error type

### ISSUE THREAD

**Issue by massimomarazzi** [laravel-filament/filament#2791]
### Package

filament/tables

### Package Version

v2.13.14

### Laravel Version

8.83.16

### Livewire Version

v2.10.5

### PHP Version

php 8.1

### Bug description

Visible method called by BulkAction get error.
Visible methot called by Action work. 

### Steps to reproduce

BulkAction::make('delete')
                ->action(fn(Collection $records) => $records->each->delete())
                ->deselectRecordsAfterCompletion()
                ->color('danger')
                ->icon('heroicon-o-trash')
                ->visible(fn($records) => $this->allowDelete())

### Relevant log output

```shell
Filament\Tables\Actions\BulkAction::Filament\Tables\Actions\{closure}(): Return value must be of type Illuminate\Database\Eloquent\Collection, null returned (View: /shared/httpd/sitemanagerpro/vendor/filament/tables/resources/views/index.blade.php)
```

**Comment by danharrin** [comment]
Please share the Flare exception URL

**Comment by massimomarazzi** [comment]
Filament\Tables\Actions\BulkAction::Filament\Tables\Actions\{closure}
vendor/filament/tables/src/Actions/BulkAction.php:63

**Comment by danharrin** [comment]
Click the "share" button on the Laravel exception page and send the link here please

**Comment by massimomarazzi** [comment]
https://flareapp.io/share/oPRM19Qm

**Comment by G3z** [comment]
this is what I found: when adding `->visible` the method `getRecords()` inside the `BulkAction` class returns `null` while return type of the closure inside `getDefaultEvaluationParameters()` is expected to be a `Collection`

**Comment by danharrin** [comment]
Fixed by #2806
